campaignion_expiry/src/SubmissionCron.php

```diff
@@ -90,15 +90,18 @@ protected function expireSubmissionBatch($up_to_time, $last_sid = 0, $batch_size
 FROM {webform_submitted_data} wsd
   INNER JOIN {webform_component} wc USING(nid, cid)
   INNER JOIN {tmp_pseudo_addresses} USING(nid, sid)
-WHERE wc.form_key<>'email'
+WHERE wc.form_key<>'email' AND wc.form_key<>'donation_amount'
 SQL;
     db_query($sql_delete_submitted_data);
 
     $sql_anonymize_email = <<<SQL
 UPDATE {webform_submitted_data} wsd
+  INNER JOIN {webform_component} wc USING(nid, cid)
   INNER JOIN {tmp_pseudo_addresses} t USING(nid, sid)
 SET wsd.data=t.email
-WHERE wsd.data NOT LIKE '%@deleted' AND wsd.data NOT LIKE '%@form-submission';
+WHERE wsd.data NOT LIKE '%@deleted'
+  AND wsd.data NOT LIKE '%@form-submission'
+  AND wc.form_key NOT LIKE 'donation_amount';
```

Comment:
This should only update email addresses, not “everything except donation amounts”. So I’d propose:

```sql
AND wc.form_key='email'
```

Reply:
good catch! thank you for the review :city_sunset: